src/socialMediaActivity.js

```diff
@@ -65,82 +66,61 @@ function registerPlatformListener(platform, eventType, blockingType, callback) {
-    }
-    if (events.includes("react") || events.includes("<all_events>")) {
-        registerPlatformListener("facebook", "react", blocking ? "blocking" : "nonblocking", callback);
-    }
-    if (events.includes("comment") || events.includes("<all_events>")) {
-        registerPlatformListener("facebook", "comment", blocking ? "blocking" : "nonblocking", callback);
-    }
-    fbPostContentInit();
+function notifyFilter(listener, listenerArguments, options) {
+    const reportedEvent = listenerArguments[0];
+    const ret = reportedEvent['platform'] == options['platform'] &&
+        options['eventTypes'].includes(reportedEvent['eventType']);
+    return ret;
 }
 
 /**
- * Register a callback for specific Reddit events. Supported events are "post", "comment",
- * "postVote", and "commentVote".
- * @param callback - the function to call when the event happens
- * @param [String] events - array of events to be tracked
- * @param blocking - whether the listener should be blocking. Allows canceling the event.
+ * A callback function for the social media activity event.
+ * @callback socialMediaActivityCallback
+ * @param {SocialMediaShareDetails} details - Additional information about the social media activity event.
+ */
+
+/**
+ * Options when adding a social media activity event listener.
+ * @typedef {Object} socialMediaActivityOptions
+ * @property {string} platform - The platform to add a listener for ('facebook', 'twitter', or 'reddit')
+ * @property {Array<string>} eventTypes - The events on the platform to notify for. For facebook, can
+ *   be 'post', 'comment', 'react', or 'reshare'. For twitter, 'tweet', 'retweet', 'favorite'. Reddit,
+ *   'post', 'comment', 'postVote', 'commentVote'.
+ * @property {boolean} blocking - Optional, set to true to allow blocking the events.
  */
-export function registerRedditActivityTracker(
-    callback,
-    events,
-    blocking = false) {
-    if (events.includes("post") || events.includes("<all_events>")) {
-        registerPlatformListener("reddit", "post", blocking ? "blocking" : "nonblocking", callback);
-    }
-    if (events.includes("comment") || events.includes("<all_events>")) {
-        registerPlatformListener("reddit", "comment", blocking ? "blocking" : "nonblocking", callback);
-    }
-    if (events.includes("postVote") || events.includes("<all_events>")) {
-        registerPlatformListener("reddit", "postVote", blocking ? "blocking" : "nonblocking", callback);
-    }
-    if (events.includes("commentVote") || events.includes("<all_events>")) {
-        registerPlatformListener("reddit", "commentVote", blocking ? "blocking" : "nonblocking", callback);
-    }
-}
+
+/**
+ * @type {events.event<socialMediaActivityCallback, socialMediaActivityOptions}
+ */
+export const onSocialMediaActivity = events.createEvent({
+    addListenerCallback: addListener,
+    notifyListenersCallback: notifyFilter
+});
```

Comment:
A big-picture design question: do we want one event for all the platforms, or one event per platform? The advantage of the former is we avoid some repetition. The advantage of the latter is library users might shoot themselves in the foot by mixing and matching platforms and activity types.